plugin/api.go

```diff
@@ -306,6 +306,17 @@ type API interface {
 	GetReactions(postId string) ([]*model.Reaction, *model.AppError)
 
 	// SendEphemeralPost creates an ephemeral post.
+	// To create an ephemeral post that can be overriden by the webapp, pass a 'type' prop within the post struct.
```

Comment:
My gut feel is that this is the wrong place to document this behaviour -- not all ephemeral posts will be overridden with a custom rendering style. And we'd have to duplicate this to `CreatePost` in case someone was looking there.

One option might be to document it in the client registry (e.g. [registerPostTypeComponent](https://github.com/mattermost/mattermost-webapp/blob/master/plugins/registry.js#L129)), but that has the same duplication problem when we eventually support mobile plugins, and it feels weird to show how to write Go code in JavaScript.

What if we started a "common workflows" document that serves as a higher-level introduction to common patterns like this? Hosted at, say, https://developers.mattermost.com/extend/plugins/workflows?

Reply:
> My gut feel is that this is the wrong place to document this behaviour -- not all ephemeral posts will be overridden with a custom rendering style.

Agreed. The override of the ephemeral posts would fall into the webApp portion of a plugin. I agree that it needs to be mentioned in the documentation of [registerPostTypeComponent](https://developers.mattermost.com/extend/plugins/webapp/reference/#registerPostTypeComponent) that the PostTypes registered can also apply for ephemeral posts. 

> What if we started a "common workflows" document that serves as a higher-level introduction to common patterns like this?

I do like overall the idea of "Common Workflows" section. However, for this particular case I feel the basic example [documentation](https://developers.mattermost.com/extend/plugins/webapp/reference/#example) does a good job of showing how to override posts and this change just expands to include ephemeral posts.

Reply:
Awesome feedback, thank you.  

I vote we close this PR.  I created two more to address [registerPostTypeComponent](https://developers.mattermost.com/extend/plugins/webapp/reference/#registerPostTypeComponent) and the example [documentation](https://developers.mattermost.com/extend/plugins/webapp/reference/#example).

* https://github.com/mattermost/mattermost-webapp/pull/3024
* https://github.com/mattermost/mattermost-developer-documentation/pull/312